android-components/components/browser/menu2/src/test/java/mozilla/components/browser/menu2/ext/PopupWindowTest.kt

```diff
@@ -126,35 +137,200 @@ class PopupWindowTest {
+        doReturn(containerHeight).`when`(menuContentView).measuredHeight
+
+        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+
+        verify(popupWindow).height = containerHeight
+        verify(popupWindow).showAsDropDown(anchor, x, -containerHeight)
+    }
+
+    @Test
+    fun `WHEN up menu container height doesn't fit within max app height & has less than required height THEN show popup up with calculated height`() {
+        val mockRecyclerView = createMockRecyclerViewWith(itemCount = 7, recyclerViewHeight = 50)
+        doReturn(mockRecyclerView).`when`(menuContentView)
+            .findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+
+        val x = 0
+        anchor = createMockViewWith(x, 40)
+        val containerHeight = mockRecyclerView.measuredHeight.plus(1)
+        doReturn(containerHeight).`when`(menuContentView).measuredHeight
+
+        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+
+        // The configuration will force the 'number of items that fit exactly with overflow' to
+        // use the minimum number of menu items.
+        val expectedMenuContainerHeight = 39
+        verify(popupWindow).height = expectedMenuContainerHeight
+        verify(popupWindow).showAsDropDown(anchor, x, -expectedMenuContainerHeight)
+    }
+
+    @Test
+    fun `WHEN up menu container height doesn't fit within max app height & at the required height THEN show popup up with calculated height`() {
+        val mockRecyclerView = createMockRecyclerViewWith(8, recyclerViewHeight = 50)
+        doReturn(mockRecyclerView).`when`(menuContentView)
+            .findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+
+        val x = 0
+        anchor = createMockViewWith(x, 40)
+        val containerHeight = mockRecyclerView.measuredHeight.plus(1)
+        doReturn(containerHeight).`when`(menuContentView).measuredHeight
+
+        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+
+        val expectedMenuContainerHeight = 34
+        verify(popupWindow).height = expectedMenuContainerHeight
+        verify(popupWindow).showAsDropDown(anchor, x, -expectedMenuContainerHeight)
+    }
+
+    @Test
+    fun `WHEN up menu container height doesn't fit within max app height & has more than required height THEN show popup up with calculated height`() {
+        val mockRecyclerView = createMockRecyclerViewWith(itemCount = 9, recyclerViewHeight = 50)
+        doReturn(mockRecyclerView).`when`(menuContentView)
+            .findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+
+        val x = 0
+        anchor = createMockViewWith(x, 40)
+        val containerHeight = mockRecyclerView.measuredHeight.plus(1)
+        doReturn(containerHeight).`when`(menuContentView).measuredHeight
+
+        popupWindow.displayPopup(menuContentView, anchor, Orientation.UP)
+
+        val expectedMenuContainerHeight = 33
```

Comment:
I wonder if there is a way not to hardcore the expected height? At least an explanation of the math, it required debugging to understand where the expected value comes from.
The calculation is something like this
`val expectedMenuContainerHeight = (itemHeight * ((viewHeight / itemHeight) - 0.5)).toInt() + viewPadding`